hazelcast-client/src/main/java/com/hazelcast/client/map/impl/nearcache/ClientHeapNearCache.java

```diff
@@ -85,11 +86,12 @@ public ClientHeapNearCache(String mapName, ClientContext context, NearCacheConfi
         if (inMemoryFormat != InMemoryFormat.BINARY && inMemoryFormat != InMemoryFormat.OBJECT) {
             throw new IllegalArgumentException("Illegal in-memory-format: " + inMemoryFormat);
         }
-        this.maxSize = nearCacheConfig.getMaxSize();
+        EvictionConfig evictionConfig = nearCacheConfig.getEvictionConfig();
+        this.maxSize = evictionConfig.isSizeConfigured() ? evictionConfig.getSize() : NearCacheConfig.DEFAULT_MAX_SIZE;
```

Comment:
This is the fallback to the old `DEFAULT_MAX_SIZE`, if the size was not configured explicitly.